### ISSUE TITLE
crossgen.exe can't generate code for methods when the type has static Vector members

### ISSUE THREAD

**Issue by rynowak** [dotnet/coreclr#7780]
Found investigating JIT time of `Microsoft.AspNetCore.Server.Kestrel.dll` with and without crossgen. Kestrel makes heavy use of vectors, and as a result much of the code cannot be crossgen-ed.

However we found some fairly harmless stuff like simple property accessors were not being crossgened.

@brianrob and @JohnChen0 helped me investigate this and determined that the root cause is that crossgen will not generate code for the members of a class if the class has even `Vector<>` member, even if the `Vector<>` members of that class are static. In our case `Frame` has some `static readonly` vectors as members. https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs#L27

I'm not sure the best way to prove that this is happening, so here's the raw data:

Excel Spreadsheet of JIT data from PerfView
[JitEvents.xlsx](https://github.com/dotnet/coreclr/files/549006/JitEvents.xlsx)

Console output of crossgen.exe
[crossgen.exe_kestrel.txt](https://github.com/dotnet/coreclr/files/549007/crossgen.exe_kestrel.txt)

**Comment by sivarv** [comment]
CC @CarolEidt 

Any method that uses SIMD `Vector<T>` will be excluded from Ngen for the reason that `Vector<T>` size depends on whether the underlying machine supports SSE2 or AVX2.  For example on SSE2 only capable machine, `Vector<T>` will be 16-bytes in size whereas on AVX2 machine it will be 32-bytes.  This is by design.

**Comment by rynowak** [comment]
Note that in this issue we're specifically referring to methods that don't use `Vector<T>` such as simple property accessors.

**Comment by brianrob** [comment]
Yup, came here to say what @rynowak said.  We're not concerned about pre-compilation of methods that use Vector<T>, but methods that don't use it but are in the same class should still be pre-compilable, right?

**Comment by jkotas** [comment]
In theory, it is possible. In practice, it is not as easy - the type loader operates on type granularity. It is why any `Vector<T>` field fails everything in the type.
